modules/keymaps.nix

```diff
@@ -6,6 +6,65 @@
 with lib; let
   helpers = import ../lib/helpers.nix {inherit lib;};
 
+  # Generates maps for a lua config
+  genMaps = mode: maps: let
+    action2Config = action: {
+      inherit
+        (action)
+        silent
+        expr
+        unique
+        noremap
+        script
+        nowait
+        remap
+        desc
+        ;
+    };
+    normalized =
+      builtins.mapAttrs
+      (key: action:
+        if builtins.isString action
+        then {
+          silent = false;
+          expr = false;
+          unique = false;
+          noremap = true;
+          remap = false;
+          script = false;
+          nowait = false;
+          action = action;
+          desc = "";
+        }
+        else
+          (action2Config action)
+          // {
+            action =
+              if action.lua
+              then helpers.mkRaw action.action
+              else action.action;
+          })
+      maps;
+  in
+    builtins.attrValues (builtins.mapAttrs
+      (key: action: {
+        action = action.action;
+        config =
+          lib.filterAttrs
+          (
+            _: v:
+              if isBool v
+              then v
+              else if isString v
+              then v != ""
+              else !isNull v
+          )
+          (action2Config action);
```

Comment:
Didn't we already call action2config when building normalized ? Why do we need it here too ?